app/src/main/java/com/orange/ods/app/ui/components/progress/ProgressCustomizationState.kt

```diff
@@ -48,4 +56,19 @@ class ProgressCustomizationState(
 
     val isCurrentValueSwitchEnabled
         get() = type.value != Type.Indeterminate
+
+    val determinateProgressAnimation
+        @Composable
+        get() = animateFloatAsState(
+            targetValue = determinateProgressValue.value,
+            animationSpec = tween(
+                durationMillis = if (determinateProgressValue.value == 0f) 0 else DeterminateProgressAnimDuration,
```

Comment:
Do you remember why you changed this:

```kotlin
durationMillis = DeterminateProgressAnimDuration,
```

to this?

```kotlin
durationMillis = if (determinateProgressValue.value == 0f) 0 else DeterminateProgressAnimDuration
```

Reply:
Yes, cause if we remove that and switch from determinate / indeterminate / determinate quickly, the animation is not replayed from start.